### DISCUSSION TITLE
Force mapstruct to use specific mapper for field?

### DISCUSSION THREAD

**Discussion by RiseOfDeath** [mapstruct/mapstruct#3139] [Q&A] [answered]
If my object have fiels of type "CustomClass" I can write special mapper "CustomClass to any what I want" and use it automaticly via "uses = {CustomClassMapper.class}".
But how can I do something like this for usual java classes, like String/Integer and etc?
f.e. source object have field of type Integer, and I want to map it to String by some specific algorythm implemented in special mapper class. (I have table with ID to Human-readable text mapping as example)

Upd.
I don't like to implement "special mapper" as static class and invoke it via expression, I want to inject it via constructor (like I can do it with "CustomClassMapper"), and use (ok if via expression, but I want to have link to it as mapper's field)

**Comment by dkellenb** [comment, marked as answer]
You can use the `@Named` annotation:

```java
@Mapper
interface MyMapper {

  @Mapping(source = "mySourceField", target = "myDestField", qualifiedByName = "myCustomNamedMappingFunction"
  DestinationObject map(SourceObject source);
  
  @Named("myCustomNamedMappingFunction")
  default String makeHumanReadable(Id id) {
     return id.getHumanReadableId();
  }

}
```

The  `@Named` annotated method can also be in another class injected via constructor. In spring it can even be an ordinary bean. I use this to transparently encrypt and decrypt ids.
